openmmtools/alchemy.py

```diff
@@ -1787,56 +1977,101 @@ def _alchemically_modify_NonbondedForce(self, reference_force, alchemical_region
         """
         # TODO Change softcore_beta to a dimensionless scalar to multiply some intrinsic length-scale, like Lennard-Jones alpha.
         # TODO Try using a single, common "reff" effective softcore distance for both Lennard-Jones and Coulomb.
-
+        forces_by_lambda = {}
+        all_alchemical_atoms = set()
+        populated_region_ids = []
+        non_pairs = []
+        for region_id, alchemical_region in enumerate(alchemical_regions):
+            if not len(alchemical_region.alchemical_atoms) == 0:
+                non_pairs.append((region_id, region_id))
+                populated_region_ids.append(region_id)
+                all_alchemical_atoms.update(alchemical_region.alchemical_atoms)
         # Don't create a force if there are no alchemical atoms.
-        if len(alchemical_region.alchemical_atoms) == 0:
+        if len(all_alchemical_atoms) == 0:
             return {'': [copy.deepcopy(reference_force)]}
 
-        # --------------------------------------------------
-        # Determine energy expression for all custom forces
-        # --------------------------------------------------
+        region_permus = non_pairs
+        if len(populated_region_ids) > 1:
+            for alchemical_region in alchemical_regions:
+                if alchemical_region.region_name == None:
+                    raise ValueError('For a alchemical calculation with multiple regions interacting via Nonbonded Forces '
+                                     'all regions populated with atoms should be named')
```

Comment:
I think we can make this requirement stricter (i.e. the exception is raised always when there are multiple unnamed alchemical regions) and move this check in `create_alchemical_system()`.